src/pymor/reductors/era.py

```diff
@@ -0,0 +1,237 @@
+from pymor.core.cache import cached, CacheableObject
+from pymor.models.iosys import LTIModel
+from pymor.operators.interface import Operator
+from pymor.operators.numpy import NumpyHankelOperator, NumpyMatrixOperator
+
+
+class ERAReductor(CacheableObject):
+    r"""Eigensystem Realization Algorithm reductor.
+
+    Constructs a (reduced) realization from a sequence of Markov parameters :math:`h_i`,
+    for :math:`i\in\{1,\,\dots,\,2s-1\}`, :math:`s\in\mathbb{N}`, by a (reduced) orthogonal
+    factorization of the Hankel matrix of Markov parameters
+
+    .. math::
+        H =
+        \begin{bmatrix}
+            h_1 & h_2 & \dots & h_s \\
+            h_2 & h_3 & \dots & h_{s+1}\\
+            \vdots & \vdots & \ddots & \vdots\\
+            h_s & h_{s+1} & \dots & h_{2s-1}
+        \end{bmatrix}=U\Sigma V^T\in\mathbb{R}^{ps\times ms},
+
+    where :math:`r\leq\min\{ms,ps\}` is the reduced order. See :cite:`K78`.
+
+    In order for the identified model to be stable, the Markov parameters decay substantially within
+    :math:`s` samples. Stability is enforced automatically through zero-padding and can be
+    deactivated by setting `force_stability=False`.
+
+    For a large number of inputs and/or outputs, the factorization of the Hankel matrix can be
+    accelerated by tangentially projecting the Markov parameters to reduce the dimension of
+    the Hankel matrix, i.e.
+
+    .. math::
+        \hat{h}_i=W_1^Th_iW_2,
+
+    where :math:`l_1\leq p` and :math:`l_2\leq m` are the number of left and right tangential
+    directions and :math:`W_1\in\mathbb{R}^{p\times l_1}` and :math:`W_2\in\mathbb{R}^{m\times l_2}`
+    are the left and right projectors, respectively. See :cite:`KG16`.
+
+    Attributes
+    ----------
+    data
+        |NumPy array| that contains the first :math:`n` Markov parameters of an LTI system.
+        Has to be one- or three-dimensional with either::
+
+            data.shape == (n,)
+
+        for scalar-valued Markov parameters or::
+
+            data.shape == (n, p, m)
+
+        for matrix-valued Markov parameters of dimension :math:`p\times m`, where
+        :math:`m` is the number of inputs and :math:`p` is the number of outputs of the system.
+    sampling_time
+        A number that denotes the sampling time of the system (in seconds).
+    force_stability
+        Whether the Markov parameters are zero-padded to double the length in order to enforce
+        Kung's stability assumption. See :cite:`K78`. Defaults to `True`.
+    feedthrough
+        (Optional) |Operator| or |Numpy array| of shape `(p, m)`. The zeroth Markov parameter that
+        defines the feedthrough of the realization. Defaults to `None`.
+    """
+
+    cache_region = 'memory'
+
+    def __init__(self, data, sampling_time, force_stability=True, feedthrough=None):
+        assert sampling_time >= 0
+        assert feedthrough is None or isinstance(feedthrough, np.ndarray) or isinstance(feedthrough, Operator)
```

Comment:
[`isinstance`](https://docs.python.org/3/library/functions.html?highlight=isinstance#isinstance) can accept a tuple of types.

```suggestion
        assert feedthrough is None or isinstance(feedthrough, (np.ndarray, Operator))
```